src/components/binding/binding.js

```diff
@@ -140,25 +149,72 @@ genotet.BindingView = function(viewName, params) {
     .on('genotet.addTrack', function() {
       var track = this.data.tracks.slice(-1).pop();
       this.loader.loadFullTrack(this.data.tracks.length, track.fileName,
-        this.data.chr);
+        this.data.chr, true);
     }.bind(this))
     .on('genotet.removeTrack', function(event, trackIndex) {
       this.data.tracks.splice(trackIndex, 1);
       this.renderer.render();
       this.panel.updateTracks();
     }.bind(this))
-    .on('genotet.gene', function(event, data) {
+    .on('genotet.updateTrack', function(event, data) {
       this.data.tracks[data.trackIndex].fileName = data.fileName;
       this.loader.loadFullTrack(data.trackIndex, data.fileName,
-        this.data.chr);
+        this.data.chr, false);
+    }.bind(this))
+    .on('genotet.loadBindingList', function() {
+      this.loader.loadBindingList();
     }.bind(this));
 
   $(this.loader)
     .on('genotet.chr', function(event, chr) {
       this.panel.updateChr(chr);
     }.bind(this))
-    .on('genotet.track', function() {
+    .on('genotet.addPanelTrack', function() {
       this.panel.updateTracks();
+    }.bind(this))
+    .on('genotet.updateTrackList', function() {
+      this.panel.updateTrackList();
+    }.bind(this));
+
+  // Set up link callbacks.
+  $(this)
+    .on('genotet.updateTrack', function(event, data) {
+      var sourceGene = data.id.split(',')[0];
```

Comment:
You can annotate that it is either node or edge using OR operator '|'.

On Friday, February 5, 2016, Liana Yang <notifications@github.com> wrote:

> In src/components/binding/binding.js
> <https://github.com/ViDA-NYU/genotet/pull/78#discussion_r52081417>:
>
> >        this.panel.updateTracks();
> > +    }.bind(this))
> > +    .on('genotet.updateTrackList', function() {
> > +      this.panel.updateTrackList();
> > +    }.bind(this));
> > +
> > +  // Set up link callbacks.
> > +  $(this)
> > +    .on('genotet.updateTrack', function(event, data) {
> > +      var sourceGene = data.id.split(',')[0];
>
> Here data could be either node or edge. I don't think annotate data is
> appropriate here. Prefer comments.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/78/files#r52081417>.
>